packages/react/src/number-field/utils/parse.ts

```diff
@@ -147,9 +147,19 @@ export function parseNumber(
 
   let groupRegex: RegExp | null = null;
   if (group) {
+    const isSpaceGroup = /\p{Zs}/u.test(group);
+    const isApostropheGroup = group === "'" || group === '’';
```

Comment:
Add a regression test where this fails otherwise?

Reply:
It depends on the browser / env settings, I think. It failed consistently on my laptop.